plutus-ledger-api/src/Plutus/V1/Ledger/EvaluationContext.hs

```diff
@@ -8,39 +10,194 @@ module Plutus.V1.Ledger.EvaluationContext
+  | LessThanInteger'cpu'arguments'intercept
+  | LessThanInteger'cpu'arguments'slope
+  | LessThanInteger'memory'arguments
+  | ListData'cpu'arguments
+  | ListData'memory'arguments
+  | MapData'cpu'arguments
+  | MapData'memory'arguments
+  | MkCons'cpu'arguments
+  | MkCons'memory'arguments
+  | MkNilData'cpu'arguments
+  | MkNilData'memory'arguments
+  | MkNilPairData'cpu'arguments
+  | MkNilPairData'memory'arguments
+  | MkPairData'cpu'arguments
+  | MkPairData'memory'arguments
+  | ModInteger'cpu'arguments'constant
+  | ModInteger'cpu'arguments'model'arguments'intercept
+  | ModInteger'cpu'arguments'model'arguments'slope
+  | ModInteger'memory'arguments'intercept
+  | ModInteger'memory'arguments'minimum
+  | ModInteger'memory'arguments'slope
+  | MultiplyInteger'cpu'arguments'intercept
+  | MultiplyInteger'cpu'arguments'slope
+  | MultiplyInteger'memory'arguments'intercept
+  | MultiplyInteger'memory'arguments'slope
+  | NullList'cpu'arguments
+  | NullList'memory'arguments
+  | QuotientInteger'cpu'arguments'constant
+  | QuotientInteger'cpu'arguments'model'arguments'intercept
+  | QuotientInteger'cpu'arguments'model'arguments'slope
+  | QuotientInteger'memory'arguments'intercept
+  | QuotientInteger'memory'arguments'minimum
+  | QuotientInteger'memory'arguments'slope
+  | RemainderInteger'cpu'arguments'constant
+  | RemainderInteger'cpu'arguments'model'arguments'intercept
+  | RemainderInteger'cpu'arguments'model'arguments'slope
+  | RemainderInteger'memory'arguments'intercept
+  | RemainderInteger'memory'arguments'minimum
+  | RemainderInteger'memory'arguments'slope
+  | Sha2_256'cpu'arguments'intercept
+  | Sha2_256'cpu'arguments'slope
+  | Sha2_256'memory'arguments
+  | Sha3_256'cpu'arguments'intercept
+  | Sha3_256'cpu'arguments'slope
+  | Sha3_256'memory'arguments
+  | SliceByteString'cpu'arguments'intercept
+  | SliceByteString'cpu'arguments'slope
+  | SliceByteString'memory'arguments'intercept
+  | SliceByteString'memory'arguments'slope
+  | SndPair'cpu'arguments
+  | SndPair'memory'arguments
+  | SubtractInteger'cpu'arguments'intercept
+  | SubtractInteger'cpu'arguments'slope
+  | SubtractInteger'memory'arguments'intercept
+  | SubtractInteger'memory'arguments'slope
+  | TailList'cpu'arguments
+  | TailList'memory'arguments
+  | Trace'cpu'arguments
+  | Trace'memory'arguments
+  | UnBData'cpu'arguments
+  | UnBData'memory'arguments
+  | UnConstrData'cpu'arguments
+  | UnConstrData'memory'arguments
+  | UnIData'cpu'arguments
+  | UnIData'memory'arguments
+  | UnListData'cpu'arguments
+  | UnListData'memory'arguments
+  | UnMapData'cpu'arguments
+  | UnMapData'memory'arguments
+  | VerifyEd25519Signature'cpu'arguments'intercept
+  | VerifyEd25519Signature'cpu'arguments'slope
+  | VerifyEd25519Signature'memory'arguments
+    deriving stock (Eq, Ord, Enum, Ix, Bounded, Generic)
+    deriving anyclass (IsParamName)
+
+mkEvaluationContext :: MonadError CostModelApplyError m => [Integer] -> m EvaluationContext
```

Comment:
So I was actually thinking we'd ask for `Map ParamName Integer`. Since we don't have a `Show` instance I think it's hard for them to abuse the names, which seems good.

But what's the difference? In some senses you can't do much with a `Map ParamName Integer` than with a `[Integer]`. I guess it means that on the ledger side because they would know about the `ParamName` type they can e.g. do sanity checks that they have all the parameters, rather than waiting for us to do it. Unsure if that's useful enough.

Reply:
@bezirg what do you think about this comment?